tests/integration/test_dashNgl.py

```diff
@@ -0,0 +1,215 @@
+import glob
+import time
+
+import dash
+from dash.dependencies import Input, Output, State
+import dash_html_components as html
+import dash_bio
+
+from common_features import simple_app_layout
+
+_COMPONENT_ID = "test-ngl"
+data_path = "tests/dashbio_demos/dash-ngl/data/"
+
+color_list = ["red", "blue"]
+
+# PDB examples
+dropdown_options = [
+    "1PNK",
+    "6CHG",
+    "3K8P",
+]
+
+# Placeholder which is loaded if no molecule is selected
+data_dict = {
+    "selectedValue": "placeholder",
+    "chain": "ALL",
+    "color": "#e41a1c",
+    "filename": "placeholder",
+    "ext": "",
+    "config": {"type": "text/plain", "input": ""},
+}
+
+
+# Helper function to load the data
+def get_data(selection, pdb_id, color):
+
+    chain = "ALL"
+
+    # Check if only one chain should be shown
+    if "." in pdb_id:
+        pdb_id, chain = pdb_id.split(".")
+
+    if pdb_id not in dropdown_options:
+        return data_dict
+    else:
+        fname = [f for f in glob.glob(data_path + pdb_id + ".*")][0]
+
+        with open(fname, "r") as f:
+            contents = f.read()
+
+        return {
+            "selectedValue": selection,
+            "chain": chain,
+            "color": color,
+            "filename": fname.split("/")[-1],
+            "ext": fname.split(".")[-1],
+            "config": {"type": "text/plain", "input": contents},
+        }
+
+
+viewer = html.Div(
+    [dash_bio.DashNgl(
+        id=_COMPONENT_ID,
+        data=data_dict)],
+    style={
+        "display": "inline-block",
+        "width": "calc(100% - 500px)",
+        "float": "left",
+        "marginTop": "50px",
+        "marginRight": "50px",
+    },
+)
+
+
+# Based on simple_app_callback
+# simple_app_callback does not work because:
+# 1.DashNgl does not accept the output as a string
```

Comment:
Could you elaborate on this? Does it require a dict? If so, you can stringify/unstringify like we've done here: https://github.com/plotly/dash-bio/blob/447ebbe01d716a0d652e744d8fe44b781fa66ad4/tests/integration/test_circos.py#L40

Reply:
yeah but this runs for "ages" because then it stringifies the whole pdb file and paste it in the input box

Reply:
Could you use a smaller file? e.g., https://github.com/plotly/dash-bio/blob/28d37e06d89d271297c526191233419b4893c034/tests/integration/test_molecule3d.py#L16

Reply:
I could use a smaller one for the first test but I need for the following tests
a file which has multiple chains and the cif files have even more text.
& for testing several molecules next to each other I need even more data
so stringifiy everything is not really an option.

Especially for the latter one I even would be forced to clear the box in between.
So this work around was the best I could hope for.

But I can assure you I tried for hours to make it work with your callbacks ;D

Reply:
OK I see! Yes, please rewrite the first test with a smaller file and keep the other tests the same.

Reply:
Ok I can give it a try!

Reply:
I tried it with your file. However, even this file has around 1000 lines. Which means quite some wait time. Haven't found the time yet to let it run completely
but maybe if you have some spare computing resources you can give it a go
see code below:
```
 # Placeholder which is loaded if no molecule is selected
data_dict = {
       "selectedValue": "placeholder",
       "chain": "ALL",
       "color": "#e41a1c",
       "filename": "placeholder",
       "ext": "",
       "config": {"type": "text/plain", "input": ""},
   }

viewer = html.Div(
    [dash_bio.NglMoleculeViewer(
        id=_COMPONENT_ID,
        data=data_dict)],
    style={
        "display": "inline-block",
        "width": "calc(100% - 500px)",
        "float": "left",
        "marginTop": "50px",
        "marginRight": "50px",
    },
)

def test_dbdn_003_show_oneMolecule_pdb(dash_duo):
   data_path =  'tests/dashbio_demos/dash-ngl-moleculeviewer/data/1BNA.pdb'

   fname = '1BNA.pdb'
   d = data_dict.copy() #shallow copy
   d['fname'] = fname
   d['selectedValue'] = fname.split('.')[0]
   d['ext'] = fname.split('.')[1]

   with open(data_path+fname, "r") as f:
      d['config']['input'] = f.read()

   app = dash.Dash(__name__)

   app.layout = html.Div(simple_app_layout(
        viewer,
    ))

   simple_app_callback(
        app,
        dash_duo,
        component_id=_COMPONENT_ID,
        test_prop_name='data',
        test_prop_value=json.dumps(d),
        prop_value_type='dict'
    )
```